src/zkp/pimod.rs

```diff
@@ -26,38 +40,38 @@ static LAMBDA: usize = crate::parameters::SOUNDNESS_PARAMETER;
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub(crate) struct PiModProof {
-    w: BigNumber,
+    random_jacobi_one: BigNumber,
     // (x, a, b, z),
     elements: Vec<PiModProofElements>,
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub(crate) struct PiModProofElements {
-    x: BigNumber,
-    a: usize,
-    b: usize,
-    z: BigNumber,
-    y: BigNumber,
+    fourth_root_y: BigNumber,
+    first_param_satisfy_x: usize,
+    second_param_satisfy_x: usize,
+    y_totient_variable: BigNumber,
+    combine_a_b: BigNumber,
```

Comment:
Hm, these names reference `x` and `y` but we renamed `x` and `y` to other things.

`y` ("combine_a_b") is the challenge; I'd recommend calling it that. The notation is a little different from other proofs but it's logically the same.

`x` is the fourth root of `y'`, but that's not the same as `y`. I don't mind calling it `fourth_root` but I'd take out the `_y` because it's not quite true.

Looking through the soundness proof, it seems like `a` and `b` are "Legendre symbols" of the fourth root with respect to the factors `p` and `q`, respectively. We could call them, like, `legendre_symbol_p`, but I'm not sure if I love that name.

`z` is a function of the challenge; it's loosely explained in the "honest verifier zero-knowledge" section but without much intuition unfortunately. I'm not sure what the stuff in the exponent does and I'm struggling to come up with a meaningful name because I still don't follow what it contributes to. In verification we compute that `z^N = y mod N`, but I'm not sure which of the relations that helps prove (e.g. does it show that the gcd is 1? that N is the product of two primes? that the primes are 3 mod 4?) I would suspect the first but I'm not sure.

Any thoughts?